spirv_msl.cpp

```diff
@@ -271,13 +271,14 @@ void CompilerMSL::build_implicit_builtins()
 	     active_input_builtins.get(BuiltInInstanceIndex) || active_input_builtins.get(BuiltInBaseInstance));
 	bool need_local_invocation_index = msl_options.emulate_subgroups && active_input_builtins.get(BuiltInSubgroupId);
 	bool need_workgroup_size = msl_options.emulate_subgroups && active_input_builtins.get(BuiltInNumSubgroups);
+	bool needs_helper_invocation_ = needs_helper_invocation || active_input_builtins.get(BuiltInHelperInvocation);
```

Comment:
Why do you need to create a new variable here? I would expect that needs_helper_invocation could just be set to true if active_input_builtins has HelperInvocation. What's the rationale for doing it this way?

Reply:
Mostly just me being scared to touch global variables.

Looked through all the uses and almost all of them used `needs_helper_invocation || active_input_builtins.get(BuiltInHelperInvocation)` anyways, so I switched it over and cleaned those up while I was at it.